sys/net/gnrc/transport_layer/tcp/gnrc_tcp_pkt.c

```diff
@@ -0,0 +1,434 @@
+        return 0;
+    }
+
+    /* Assign pkt and increase users: every send attempt consumes a user */
+    tcb->pkt_retransmit = pkt;
+    gnrc_pktbuf_hold(pkt, 1);
+
+    /* RTO Adjustment */
+    if (!retransmit) {
+        /* If this is the first transmission: rto is 1 sec (Lower Bound) */
+        if (tcb->srtt == GNRC_TCP_RTO_UNINITIALIZED || tcb->rtt_var == GNRC_TCP_RTO_UNINITIALIZED) {
+            tcb->rto = GNRC_TCP_RTO_LOWER_BOUND;
+        }
+        else {
+            tcb->rto = tcb->srtt + MAX(GNRC_TCP_RTO_GRANULARITY,  GNRC_TCP_RTO_K * tcb->rtt_var);
+        }
+    }
+    else {
+        /* If this is a retransmission: Double the rto (Timer Backoff) */
+        tcb->rto *= 2;
+
+        /* If the transmission has been tried five times, we assume srtt and rtt_var are bogus */
+        /* New measurements must be taken */
+        if (tcb->retries >= 5) {
+            tcb->srtt = GNRC_TCP_RTO_UNINITIALIZED;
+            tcb->rtt_var = GNRC_TCP_RTO_UNINITIALIZED;
+        }
+    }
+
+    /* Perform Boundrychecks on current RTO before usage */
+    if (tcb->rto < GNRC_TCP_RTO_LOWER_BOUND) {
+        tcb->rto = GNRC_TCP_RTO_LOWER_BOUND;
+    }
+    else if (tcb->rto > GNRC_TCP_RTO_UPPER_BOUND) {
+        tcb->rto = GNRC_TCP_RTO_UPPER_BOUND;
+    }
+
+    /* Setup retransmission timer, msg to TCP thread with ptr to tcb */
+    tcb->msg_tout.type = MSG_TYPE_RETRANSMISSION;
+    tcb->msg_tout.content.ptr = (void *)tcb;
+    xtimer_set_msg(&tcb->tim_tout, tcb->rto, &tcb->msg_tout, _gnrc_tcp_pid);
+    return 0;
+}
+
+
+int8_t _pkt_acknowledge(gnrc_tcp_tcb_t* tcb, uint32_t ack)
+{
+    int32_t rtt = 0;
+    uint32_t seg = 0;
+    gnrc_pktsnip_t *snp = NULL;
+    tcp_hdr_t *hdr;
+
+    /* Retransmission Queue is empty. Nothing to ACK there */
+    if (tcb->pkt_retransmit == NULL) {
+        DEBUG("gnrc_tcp_pkt : _pkt_acknowledge() : There is no packet to ack\n");
+        return -ENODATA;
+    }
+
+    LL_SEARCH_SCALAR(tcb->pkt_retransmit, snp, type, GNRC_NETTYPE_TCP);
+    hdr = (tcp_hdr_t *) snp->data;
+
+    /* There must be a packet, waiting to be acknowledged. */
+    seg = byteorder_ntohl(hdr->seq_num) + _pkt_get_seg_len(tcb->pkt_retransmit) - 1;
+
+    /* If segment can be acknowledged -> stop timer, release packet from pktbuf and update rto. */
+    if (LSS_32_BIT(seg, ack)) {
+        xtimer_remove(&(tcb->tim_tout));
+        gnrc_pktbuf_release(tcb->pkt_retransmit);
+        tcb->pkt_retransmit = NULL;
+
+        /* Measure Round Trip Time */
+        rtt = xtimer_now() - tcb->rtt_start;
+
+        /* Use sample only if ther was no timeroverflow and no retransmission (Karns Alogrithm) */
+        if (tcb->retries == 0 && rtt > 0) {
+            /* If this is the first sample taken */
+            if (tcb->srtt == GNRC_TCP_RTO_UNINITIALIZED && tcb->rtt_var == GNRC_TCP_RTO_UNINITIALIZED) {
+                tcb->srtt = rtt;
+                tcb->rtt_var = 0.5f * rtt;
```

Comment:
Actually you are write, but instead a division by 2 a simple shift should to the trick.

As far as I see it is the "First RTT measurement R is made" section. 

The "RTO <- SRTT + max (G, K*RTTVAR)" calculation happens in _setup_retransmit() because its the same formula that has to be calculated before the Retransmit timer is initialized.